src/meson.build

```diff
@@ -71,15 +71,12 @@ install_headers(headers,
 
 pkg = import('pkgconfig')
 pkg.generate(
-  version: meson.project_version(),
-  libraries: myself,
```

Comment:
Hmm, I would suspect you still want to pass the primary library as the first posarg to pkg.generate?

Reply:
Thanks for your comment.

Indeed. I overlooked that the first positional argument is available since v0.46. I think it's preferable since it's possible to parse ELF binary for libhinoko library to generate .pc file without passing explicit parameters in the latest version of meson:

https://mesonbuild.com/Pkgconfig-module.html

```
Since 0.46 a StaticLibrary or SharedLibrary object can optionally be passed as first positional argument. If one is provided a default value will be provided for all required fields of the pc file:

install_dir is set to pkgconfig folder in the same location than the provided library.
description is set to the project's name followed by the library's name.
name is set to the library's name.

Implicit dependencies
The exact rules followed to find dependencies that are implicitly added into the pkg-config file have evolved over time. Here are the rules as of Meson 0.49.0, previous versions might have slightly different behaviour.
...
```

I still wonder the reason that .pc file is generated expectedly without the argument, but later I'll force-push respined patchset which includes your indication.

Reply:
> Is it generated expectedly?
>
> I would expect it to be generated successfully, but to be missing the libhinoko library since it's not listed anywhere in the invocation.

Indeed. I did take less care of differences between before/after the simplification. I can see explicit difference there...

```
$ diff -u hinoko.pc.prev hinoko.pc.new
--- hinoko.pc.prev      2022-03-10 12:02:43.902469857 +0900
+++ hinoko.pc.new       2022-03-10 12:05:15.969281391 +0900
@@ -5,5 +5,6 @@
 Name: hinoko
 Description: A library to operate OHCI 1394 controller for isochronous stream
 Version: 0.6.0
-Requires: glib-2.0, gobject-2.0
+Requires: glib-2.0, gobject-2.0 >= 2.34.0
+Libs: -L${libdir} -lhinoko
 Cflags: -I${includedir}/hinoko
```

Just now I force-pushed and move the HEAD of PR branch into d9489cbdf9cc . I'd like you to test that it can generate as we expected.

Thank you.

Reply:
In the latest force push of this PR, the generated pkg-config file is identical before and after this PR. So that looks good.

Looking at the change itself it seems like the correct use of `pkg.generate`. :)

Incidentally, you probably don't need to explicitly list glib-2.0 since you don't explicitly link to it and gobject-2.0 already has it as a public `Requires:`

You can pass `requires: dependencies` directly, you don't need to use the string names of them, which allows you to sync the names with the dependencies that you actually look up.

Ordinarily they would be listed by default anyway even if not referenced at all, because libhinoko links to them, but... that only adds them to `Requires.private` while you do need them to be public requires.

Reply:
> In the latest force push of this PR, the generated pkg-config file is identical before and after this PR. So that looks good.
> 
> Looking at the change itself it seems like the correct use of `pkg.generate`. :)

It's nice.

> Incidentally, you probably don't need to explicitly list glib-2.0 since you don't explicitly link to it and gobject-2.0 already has it as a public `Requires:`
> 
> You can pass `requires: dependencies` directly, you don't need to use the string names of them, which allows you to sync the names with the dependencies that you actually look up.
> 
> Ordinarily they would be listed by default anyway even if not referenced at all, because libhinoko links to them, but... that only adds them to `Requires.private` while you do need them to be public requires.

To be honestly, I'm not so good at pkg-config itself as well as pkg-config module of meson. After a short research, the difference between `Requires` and `Requires.private` fields relates to `overlinking`, which brings useless rebuilt of application even in case of dynamic linking.

In case of `libhinoko`, libhinoko applications should not be rebuilt as long as built for dynamic linking when glib-2.0/gobject-2.0 is changed. I guess that the latest tools for linking supports function to prevent from overlinking issue, while for safe I conclude not to use `Require` field but to use `Require.private` field. In the meaning, it's convenient to use default behaviour of pkg-config module.

Reply:
I moved HEAD of PR branch into f20944e6822b from d9489cbdf9cc as merge candicate.

Reply:
@eli-schwartz 

Thanks for your helpful comment. If you have enough time to care of the other libraries I maintain in alsa-project, your comment to them makes me really happy. They support g-i as well as libhinoko:

 * https://github.com/alsa-project/libhinawa
 * https://github.com/alsa-project/alsa-gobject/

Regards